cumulusci/tasks/bulkdata.py

```diff
@@ -442,13 +444,18 @@ def _store_inserted_ids(self, mapping, job_id, local_ids_for_batch):
                 self.logger.info(
                     "  Updated {} for batch {}".format(id_table_name, batch_id)
                 )
-            except Exception:  # pragma: nocover
-                # If we can't download one result file,
-                # don't let that stop us from downloading the others
-                self.logger.error(
-                    "Could not download batch results: {}".format(batch_id)
-                )
-                continue
+            except Exception as e:  # pragma: nocover
+                # We can't get new Ids for some reason, or determine batch status.
+                # Fail the job to preserve integrity of data store.
+                if not isinstance(e, BulkDataException):
+                    raise BulkDataException(
+                        "Failed to download results for batch {} ({})".format(
+                            batch_id, e
+                        )
+                    )
+                else:
+                    raise e
```

Comment:
I'd suggest writing this as follows:

    except BulkDataException:
        raise
    except Exception as e:
        raise BulkDataException("Failed to download results for batch {} ({})".format(batch_id, str(e)))

Reply:
👍 Also removing the old nocover pragma and adding a test case.